android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java

```diff
@@ -53,11 +69,84 @@ public void present(@NonNull String document, @NonNull ReadableMap configuration
         }
     }
 
+    @ReactMethod
+    public void setPageIndex(final int pageIndex, final boolean animated) {
+        if (currentActivity instanceof PdfActivity) {
+            final PdfActivity activity = (PdfActivity) currentActivity;
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    if (activity.getDocument() != null) {
+                        // If the document is loaded we can instantly set the page index.
+                        activity.setPageIndex(pageIndex, animated);
+                    } else {
+                        activity.getPdfFragment().addDocumentListener(new SimpleDocumentListener() {
```

Comment:
Why is this necessary? `PdfFragment` should buffer the page index for as long as no document is loaded.

Reply:
That's true, but unless you disable restoring the last opened page, the last opened page will override any page set before the document was loaded.